packages/@react-aria/avatar/src/useAvatar.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaAvatarProps} from '@react-types/avatar';
+import {filterDOMProps, mergeProps} from '@react-aria/utils';
+import React, {ImgHTMLAttributes, RefObject} from 'react';
+import {usePress} from '@react-aria/interactions';
+
+interface AriaAvatarOptions extends AriaAvatarProps {
+  /**
+   * The HTML element used to render the avatar, e.g. 'img', or 'div'.
+   *
+   * @default 'img'
+   */
+   elementType?: string
+}
+
+interface AvatarAria {
+  /**
+   * Props for the avatar element.
+   */
+  avatarProps: ImgHTMLAttributes<HTMLElement>,
+  /**
+   * Whether the avatar is currently pressed.
+   */
+  isPressed: boolean
+}
+
+/**
+ * Provides the behavior and accessibility implementation for an avatar
+ * component. An avatar is a thumbnail representation of an entity, such as a
+ * user or an organization.
+ */
+export function useAvatar(props: AriaAvatarOptions, ref: RefObject<HTMLElement>): AvatarAria {
+  const {
+    elementType = 'img',
+    alt = '',
+    isDisabled,
+    onPress,
+    onPressStart,
+    onPressEnd,
+    // @ts-ignore
+    onClick: deprecatedOnClick,
+    ...otherProps
+  } = props;
+
+  const avatarProps = {
+    ...(elementType === 'img' && {alt}),
+    ...(elementType !== 'img' && {
+      'aria-label': alt,
+      role: 'img'
+    }),
+    tabIndex: !isDisabled ? 0 : undefined
+  };
+
+  const domProps = filterDOMProps(otherProps);
+
+  const {isPressed, pressProps} = usePress({onPress, onPressStart, onPressEnd, isDisabled, ref});
```

Comment:
Mentioning this for other opinions:
I know in a previous PR that we made a stance that we only wanted to include isPressed and other states in the aria hook if absolutely necessary. What do we think about having it here? IMO I think we might be able to drop it here